proposals/channel-jamming-mitigation.md

```diff
@@ -0,0 +1,164 @@
+# Upfront HTLC Fees
+
+## Table of Contents
+* [Proposal](#proposal)
+  * [Introduction](#introduction)
+  * [Threat Model](#threat-model)
+  * [Mitigation](#mitigation)
+    * [Quick Jamming Mitigation ](#quick-jamming-mitigation)
+    * [Slow Jamming Mitigation](#slow-jamming-mitigation)
+    * [Justification](#justification)
+* [Network Upgrade](#network-upgrade)
+* [References](#references)
+
+## Proposal
+
+### Introduction
+Channel jamming is a known denial-of-service attack against the Lightning 
+Network. An attacker that controls both ends of a payment route disrupt usage 
+of a channel by sending payments that are destined to fail through a target 
+route. This attack can exhaust the resources along the target route in two ways: 
+1. Quick Jamming: sending a continuous stream of payments through a route that
+   that are quickly resolved, but block all of the liquidity or HTLC slots 
+   along the route.
+2. Slow Jamming: sending a slow stream of payments through a route and only 
+   failing them at the latest possible timeout, blocking all of the liquidity
+   or HTLC slots along the route for the duration.
+
```

Comment:
Should there be a note about the diff between slow and quick being not rigorously defined?